File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int start = 0, end = s.length()-1;
        char[] s1 = s.toCharArray();
        while(start < end) {
            char b = s1[start];
            char e = s1[end];
            if(!Character.isAlphabetic(b) && !Character.isDigit(b)) {
                start++;
                continue;
            } else if(!Character.isAlphabetic(e) && !Character.isDigit(e)) {
                end--;
                continue;
            } 
            if(Character.isUpperCase(b)) {
                b = Character.toLowerCase(b);
            } 
            if(Character.isUpperCase(e)) {
                e = Character.toLowerCase(e);
            }

            if(b != e) {
                return false;
            }
            start++;
            end--;
        }

        return true;
    }

}

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        pointer1 = 0
        pointer2 = len(s) - 1
        while pointer1 < pointer2:
            start = s[pointer1]
            end = s[pointer2]

            if not (start.isalpha() or start.isdigit()):
                pointer1 += 1
                continue

            if not (end.isalpha() or end.isdigit()):
                pointer2 -= 1
                continue

            start = start.lower() if start.isalpha() else start
            end = end.lower() if end.isalpha() else end

            if start != end:
                return False
            pointer1 += 1
            pointer2 -= 1

        return True       